wikimedia/wikipedia-ios
# Notifications - add mark as read functionality

**Phabricator:** 
https://phabricator.wikimedia.org/T287302
https://phabricator.wikimedia.org/T288687

### Notes
This PR sets up some basic functionality of marking notifications as read or unread, or marking all notification as read via the MediaWiki notifications API. It builds on PR #4066 to trigger marking a single notification as read or unread via the swipe action sheet option. I decided to do a bit of a fire-and-forget approach on these without completion blocks, but I'm open to altering it here or later when we work on the other mark as read [task](https://phabricator.wikimedia.org/T288665).

Note that even though I am triggering one of these methods through a swipe action, the state won't update itself because the data flow [PR](https://github.com/wikimedia/wikipedia-ios/pull/4049) is not merged yet. For a preview of all PRs working together you can test the `notifications-plus-prs-2` branch.

### Test Steps
1. Fresh launch app. Login and go to Notifications Center. Also open the Notifications Center on desktop Wikipedia (https://en.wikipedia.org/wiki/Special:Notifications), and go to the "Unread" section in a browser.
2. Swipe on one of your notifications on device and see the the action sheet display. Tap the "Mark as read/unread" option.
3. When you swipe and mark something as unread, the notification should appear on this Desktop unread section after a browser refresh. When you swipe again mark something as read, the notification should disappear from this Desktop unread section after a browser refresh. 
4. Repeat step 2 to mark a few notifications as unread, ideally notifications from multiple languages or projects. Refresh in step 3 to confirm they show as unread on Desktop.
5. Add temporary hook to mark all as read when tapping the edit button [here](https://github.com/wikimedia/wikipedia-ios/blob/410766f45385a272d63a5b47e1294f0a4c368b94/Wikipedia/Code/NotificationsCenterViewController.swift#L81): `viewModel.markAllAsRead()` 
6. Rerun project and tap Edit button. Refresh desktop Wikipedia. Confirm all project notifications were marked as read.
